spyder/widgets/editor.py

```diff
@@ -2090,11 +2092,14 @@ def set_layout_settings(self, settings):
             for index, finfo in enumerate(editorstack.data):
                 editor = finfo.editor
                 # FIXME: Temporal fix
-                try:
-                    editor.go_to_line(clines[index])
-                except IndexError:
+                if dont_goto is not None:
+                    # skip go to line for first file because is already there
                     pass
-            editorstack.set_current_filename(cfname)
+                else:
+                    try:
+                        editor.go_to_line(clines[index])
+                    except IndexError:
+                        pass

```

Comment:
Ok, then please leave a `TODO` comment here. saying that this is going to be fixed in a future PR :-)